ansible/environments/distributed/group_vars/all

```diff
@@ -70,3 +70,4 @@ instances:
   - name: db
     num_instances: 1
     flavor:
+    volume:
```

Comment:
When the user first runs the provision_env_dist playbook, the Openwhisk VMs do not exist yet in the targeted IAAS, therefore they will not have a valid hosts file yet. The provision playbook iterates through the instances list, makes Openstack api calls for each instance object, and generates the hosts file dynamically based on the information returned by the api results. The num_instances and name properties are required here to define the group names that'll be placed in the hosts file, as well as the number of instances that will be in each group. This provision playbook is optional if the user wants to manage the instances and hosts file themselves.

I've removed the include_vars calls, looks like those weren't needed.
I agree, there is a need for redesign so we can get rid of those tricky set_facts calls, can probably start on that later this week. 
I've made adjustments based on your other suggestions.